### ISSUE TITLE
[xray] Ctrl-C in Cluster Mode renders cluster unusable

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#2484]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: g2.2xlarge DL ami
- **Ray installed from (source or binary)**: latest wheel
- **Python version**: 3.6
- **Exact command to reproduce**:

```
export RAY_USE_XRAY=1
ray start --head --redis-port 6379
python ray/python/ray/examples/tune_mnist_async_hyperband.py
[Ctrl-C after a bit]
# This subsequent call will hang when actors are launched
python ray/python/ray/examples/tune_mnist_async_hyperband.py
```

### Describe the problem
This behavior works fine without X-Ray.

### Log:
This was generated after `ray stop` on a separate thread.
```
/ray/src/local_scheduler/local_scheduler_client.cc:304 Check failed: static_cast<ray::protocol::MessageType>(type) == ray::protocol::MessageType::WaitReply
/home/ubuntu/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so(+0x18995)[0x7f78eb0c7995]
/home/ubuntu/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so(_ZN3ray8internal7CerrLogD1Ev+0x8e)[0x7f78eb0d0f9e]
/home/ubuntu/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so(_Z20local_scheduler_waitP24LocalSchedulerConnectionRKSt6vectorIN3ray8UniqueIDESaIS3_EEilb+0x283)[0x7f78eb0d6753]
/home/ubuntu/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so(+0x1a609)[0x7f78eb0c9609]
python(_PyCFunction_FastCallDict+0x91)[0x55f50da30f11]
python(+0x19cbec)[0x55f50dabebec]
python(_PyEval_EvalFrameDefault+0x30a)[0x55f50dae319a]
python(+0x1959a6)[0x55f50dab79a6]
python(+0x196a11)[0x55f50dab8a11]
python(+0x19ccc5)[0x55f50dabecc5]
python(_PyEval_EvalFrameDefault+0x30a)[0x55f50dae319a]
python(+0x1967db)[0x55f50dab87db]
python(+0x19ccc5)[0x55f50dabecc5]
python(_PyEval_EvalFrameDefault+0x30a)[0x55f50dae319a]
python(+0x1967db)[0x55f50dab87db]
python(+0x19ccc5)[0x55f50dabecc5]
python(_PyEval_EvalFrameDefault+0x30a)[0x55f50dae319a]
python(+0x1959a6)[0x55f50dab79a6]
python(+0x196a11)[0x55f50dab8a11]
python(+0x19ccc5)[0x55f50dabecc5]
python(_PyEval_EvalFrameDefault+0x1021)[0x55f50dae3eb1]
python(PyEval_EvalCodeEx+0x329)[0x55f50dab9529]
python(PyEval_EvalCode+0x1c)[0x55f50daba2cc]
python(+0x214af4)[0x55f50db36af4]
python(PyRun_FileExFlags+0xa1)[0x55f50db36ef1]
python(PyRun_SimpleFileExFlags+0x1c4)[0x55f50db370f4]
python(Py_Main+0x648)[0x55f50db3ac28]
python(main+0xee)[0x55f50da0271e]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f7930989830]
python(+0x1c7c98)[0x55f50dae9c98]
Aborted (core dumped)
```

**Comment by richardliaw** [comment]
This one is a little harder to reproduce an issue for...

**Comment by ericl** [comment]
This seems to happen any time you Ctrl-C or Ctrl-\ a running Ray program.

**Comment by richardliaw** [comment]
After 1 Ctrl-C: 
```
In [1]: import ray
ray.
In [2]: ray.init(redis_address="localhost:6379")
Detected environment variable 'RAY_USE_XRAY'.
/home/ubuntu/ray-1/src/common/io.cc:119: Connection to socket failed for pathname /tmp/raylet3105601
/home/ubuntu/ray-1/src/common/io.cc:127: Could not connect to socket /tmp/raylet3105601
/home/ubuntu/ray-1/python/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so(+0x16c50)[0x7f776c3b7c50]
/home/ubuntu/ray-1/python/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so(_Z22connect_ipc_sock_retryPKcil+0x2d5)[0x7f776c3e5535]
/home/ubuntu/ray-1/python/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so(_Z29LocalSchedulerConnection_initPKcRKN3ray8UniqueIDEbS4_b+0x8f)[0x7f776c3cf64f]
/home/ubuntu/ray-1/python/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so(+0x178fd)[0x7f776c3b88fd]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(+0x9de3b)[0x7f7790dd6e3b]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyObject_Call+0x43)[0x7f7790d817a3]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x3bb9)[0x7f7790e17b69]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCode+0x1a)[0x7f7790e1d70a]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x5b30)[0x7f7790e19ae0]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(+0x6d377)[0x7f7790da6377]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyObject_Call+0x43)[0x7f7790d817a3]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x250e)[0x7f7790e164be]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x6a08)[0x7f7790e1a9b8]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x7e9)[0x7f7790e1d4e9]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCode+0x1a)[0x7f7790e1d70a]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(+0xfd9cd)[0x7f7790e369cd]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyRun_FileExFlags+0x78)[0x7f7790e37b48]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(PyRun_SimpleFileExFlags+0xe8)[0x7f7790e38d68]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/../lib/libpython2.7.so.1.0(Py_Main+0xbac)[0x7f7790e4af8c]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f7790062830]
/home/ubuntu/anaconda3/envs/tensorflow_p27/bin/python(+0x8bf)[0x5568e3a828bf]
Aborted (core dumped)
```

**Comment by robertnishihara** [comment]
@richardliaw this may have been fixed by #2826. Can you check if it works now?

**Comment by richardliaw** [comment]
This is fixed!
